buffer/src/main/java/io/netty/buffer/ByteBufUtil.java

```diff
@@ -246,10 +246,13 @@ public static int indexOf(ByteBuf needle, ByteBuf haystack) {
         }
 
         // When the needle has only one byte that can be read,
-        // the firstIndexOf method needs to be called
+        // the firstIndexOf method can be used
         if (m == 1) {
-            return firstIndexOf((AbstractByteBuf) haystack, haystack.readerIndex(),
-                    haystack.writerIndex(), needle.getByte(needle.readerIndex()));
+            return haystack instanceof AbstractByteBuf ? firstIndexOf((AbstractByteBuf) haystack,
+                    haystack.readerIndex(), haystack.writerIndex(),
+                    needle.getByte(needle.readerIndex()))
+                    : haystack.indexOf(haystack.readerIndex(), haystack.writerIndex(),
+                          needle.getByte(needle.readerIndex()));
```

Comment:
Sorry, I meant instead of doing the `instanceof` check, we can just always call `haystack.indexOf(haystack.readerIndex(), haystack.writerIndex(), needle.getByte(needle.readerIndex()))`